### PR TITLE
Adds always- and never-notify modifiers to Owner class

### PR DESCRIPTION
**Pull request by rcebulko**
Also includes methods in the Owners Bot to use these modifiers to determine the set of reviewers to request and owners to notify/tag in the PR

Part of #395

### CODE REVIEW

File: owners/src/types.js [hunk 1]
```diff
@@ -63,10 +63,21 @@ let OwnersParserResult;
  */
 let FileRef;
 
+/**
+ * The result of an owners check.
+ *
+ * @typedef {{
+ *   checkRun: CheckRun,
+ *   reviewers: ?string[],
```
Comment (danielrozenberg) [inline owners/src/types.js]: Why did you choose to have this optional, as opposed to always returning a list, even when it's empty (=> less null checks)

File: owners/test/owner.test.js [hunk 2]
```diff
@@ -79,14 +111,32 @@ describe('owner teams', () => {
...
   describe('toString', () => {
     it("returns the team members' usernames as a comma-separated list", () => {
-      expect(owner.toString()).toEqual('auser, anothermember');
+      expect(owner.toString()).toEqual('[auser, anothermember]');
+    });
+
+    it('includes any modifiers', () => {
+      const owner = new TeamOwner(myTeam, OWNER_MODIFIER.NOTIFY);
+      expect(owner.toString()).toContain(' (always notify)');
+      expect(owner.toString().endsWith(' (always notify)')).toBe(true);
```
Comment (danielrozenberg) [inline owners/test/owner.test.js]: You can replace these two with
```javascript
expect(owner.toString()).toMatch(/ \(always notify\)$/);
```

File: owners/src/types.js (hunk 1 above)
Comment (rcebulko) [reply owners/src/types.js]: Sure, done.

File: owners/test/owner.test.js (hunk 2 above)
Comment (rcebulko) [reply owners/test/owner.test.js]: Whoops, didn't mean to have both in here (I think it was the result of a merge/rebase/stash or something). Regardless, I'll swap out for this matcher. Done.

**Pull request merged**